tests/migrations/test_commands.py

```diff
@@ -13,13 +13,41 @@
 from django.db.backends.base.schema import BaseDatabaseSchemaEditor
 from django.db.migrations.exceptions import InconsistentMigrationHistory
 from django.db.migrations.recorder import MigrationRecorder
-from django.test import override_settings
+from django.test import TestCase, override_settings
 
 from .models import UnicodeModel, UnserializableModel
 from .routers import TestRouter
 from .test_base import MigrationTestBase
 
 
+class AppNameErrorTests(TestCase):
+    def test_migrate_nonexistent_app_error(self):
+        msg = "No installed app with label 'this_app_does_not_exist'."
+        with self.assertRaisesMessage(CommandError, msg):
+            call_command('migrate', 'this_app_does_not_exist')
+
+    def test_migrate_app_name_with_dots(self):
```

Comment:
Now we can call this `test_migrate_app_name_specified_as_label` (and similar for the similar test).